bigquery/storage/managedwriter/managed_stream.go

```diff
@@ -140,3 +159,173 @@ func (ms *ManagedStream) Finalize(ctx context.Context) (int64, error) {
 	}
 	return resp.GetRowCount(), nil
 }
+
+// getStream returns either a valid client or permanent error.
+func (ms *ManagedStream) getStream(arc *storagepb.BigQueryWrite_AppendRowsClient) (*storagepb.BigQueryWrite_AppendRowsClient, chan *pendingWrite, error) {
+	ms.mu.Lock()
+	defer ms.mu.Unlock()
+	if ms.err != nil {
+		return nil, nil, ms.err
+	}
+	ms.err = ms.ctx.Err()
+	if ms.err != nil {
+		return nil, nil, ms.err
+	}
+
+	// Always return the retained ARC if the arg differs.
+	if arc != ms.arc {
```

Comment:
This will differ when ms.arc is nil, does this need an extra check?

Reply:
The only time ms.arc will be nil is when arc is nil (we have never opened a connection), so we'll skip this conditional.